lib/dock.rb

```diff
@@ -0,0 +1,32 @@
+class Dock
+  attr_reader :name, :max_rental_time, :currently_rented_boats,
+              :revenue, :total_hours_by_rental_type, :charges
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @currently_rented_boats = []
```

Comment:
With `@currently_rented_boats` and `@rentals_by_ccn` it seems like you are storing similar information in two places.  It may make more sense to store one `@rental_information` hash with renters as keys, and an array of the boats they have rented as values.